portal-impl/src/com/liferay/portal/servlet/filters/header/HeaderFilter.java

```diff
@@ -133,7 +131,7 @@ else if (StringUtil.equalsIgnoreCase(name, HttpHeaders.EXPIRES)) {
 			}
 		}
 
-		httpServletResponse.addHeader(name, value);
+		httpServletResponse.setHeader(name, value);
```

Comment:
@binhtran92 Sorry I missed an important thing in my previous comment -- the RFC. So I did some extra search on the internet, and it turns out that **SOME** HTTP headers are allowed to duplicate in a response.

So RFC7230, the latest RFC for HTTP 1.1, [says that](https://www.rfc-editor.org/rfc/rfc7230#section-3.2.2)

>   A sender MUST NOT generate multiple header fields with the same field
>   name in a message **unless** either the entire field value for that
>   header field is defined as a comma-separated list [i.e., #(values)]
>   or the header field is a well-known exception (as noted below).

And in RFC7234, `Cache-Control` [is defined as](https://www.rfc-editor.org/rfc/rfc7234#section-5.2):

```
     Cache-Control   = 1#cache-directive
     cache-directive = token [ "=" ( token / quoted-string ) ]
```

In which `1#cache-directive` means [a list with at least one](https://www.w3.org/Protocols/rfc2616/rfc2616-sec2.html#sec2) cache-directive.

So this means `Cache-Control` can appear multiple times in the HTTP header, as long as it can construct a meaningful list.

On the other hand, `Set-Cookie` should not be de-duplicated at all because it's a "well-know exception" in RFC7230.

For `Expires`, it should not duplicate: https://www.rfc-editor.org/rfc/rfc7234#section-5.3

Based on that, for `Cache-Control`, we should interpret it's value and merge them as a list according to the RFC (maybe also need to avoid conflicting directives?), while for `Expires`, just de-duplicate. For other headers that the `HeaderFilter` does not handle, `addHeader` them as is.